src/kernel.hpp

```diff
@@ -127,37 +135,55 @@ struct cvk_kernel_argument_values {
     }
 
     bool init() {
-        if (m_kernel->has_pod_arguments()) {
+        if (m_kernel->has_pod_buffer_arguments()) {
             auto buffer = m_kernel->allocate_pod_buffer();
             if (buffer == nullptr) {
                 return false;
             }
 
             m_pod_buffer = std::move(buffer);
+        } else if (m_kernel->has_pod_arguments()) {
+            auto buffer = m_kernel->allocate_pod_pushconstant_buffer();
+            if (buffer == nullptr) {
+                return false;
+            }
```

Comment:
An exception will have be thrown, this error handling code will never run.

Reply:
I'm not sure I follow.  Like the VkBuffer case this feeds back to return CL_OUT_OF_RESOURCES when initializing the kernel.

Reply:
The only thing that `allocate_pod_pushconstant_buffer` does is allocate host memory. If the allocation fails, a `std::bad_alloc` will have been thrown (and either caught somewhere or it will have triggered a call to `std::terminate`). This function can never return a `nullptr`. Maybe renaming the function would make this clearer.

Host memory allocation failures are currently left unhandled in clvk (see #101).

Reply:
Right, I'd forgotten about that. I will change this to return true for now then. I can add a comment to refer to that issue as well.